src/renderers/shared/stack/reconciler/ReactRef.js

```diff
@@ -56,20 +66,21 @@ ReactRef.shouldUpdateRefs = function(prevElement, nextElement) {
   // is made. It probably belongs where the key checking and
   // instantiateReactComponent is done.
 
-  var prevEmpty = prevElement === null || prevElement === false;
-  var nextEmpty = nextElement === null || nextElement === false;
-
   return (
     // This has a few false positives w/r/t empty components.
-    prevEmpty || nextEmpty ||
+    prevElement == null || prevElement === false || // prevEmpty
```

Comment:
prevElement/nextElement should never be undefined here. Can we change the types so that is enforced? `ReactElement | null | false` or something.